src/OrleansRuntime/Storage/MemoryStorageGrain.cs

```diff
@@ -119,23 +119,23 @@ private static string NewEtag()
 
             private void ValidateEtag(string currentETag, string receivedEtag, string grainStoreKey, string operation)
             {
+                // if we have no current etag, we will except the users.
+                // This is a mitigation for when the memory storage grain is lost do to silo crash.
+                if (currentETag == null)
+                    return;
```

Comment:
Good catch!

Simpler solution I think is to track delete state by setting it's etag to empty string, so we can distinguish between deleted data and non-existent data.

Tracking info in the eTag, bloats the etag, and requires us to parse the information out of it.